src/fides/lib/oauth/api/routes/user_endpoints.py

```diff
@@ -71,7 +72,11 @@ def create_user(
             status_code=HTTP_400_BAD_REQUEST, detail="Username already exists."
         )
 
-    user = FidesUser.get_by(db, field="username", value=user_data.username)
+    user = (
+        FidesUser.query(db)
+        .filter(func.lower(FidesUser.username) == user_data.username.lower())
+        .first()
+    )
```

Comment:
from the ticket:
> When implementing this, we should enforce this in the database with case-insensitivity.

These changes are good at preventing duplicate users at the code level but there's nothing at the database level.

Some ideas [here](http://shuber.io/case-insensitive-unique-constraints-in-postgres/) include changing the type of Fidesuser.username to a `citext` type or adding a unique index instead of a constraint.  I'd look more into the pros and cons of these approaches.

Reply:
I converted it to use citext instead. Thanks for the suggestion!